docs/source/community/pull_requests.rst

```diff
@@ -0,0 +1,360 @@
+and reviewers often learn new techniques or discover previously unknown
+features by reviewing code.
+
+PRs are also about staking a claim. When you open a draft PR you're telling
+people that you are going to work on resolving a bug etc. This tells other
+people not to duplicate the effort and/or gives them a chance to join the
+cause.
+
+
+*****************
+PR Considerations
+*****************
+
+This section lists considerations for authors and/or reviewers of PRs.
+Following semantic versioning, we can think of each PR as addressing one of
+three use cases:
+
+#. API breaking changes
+
+   - Any change to a public-facing API which breaks previously working code.
+
+#. New features
+
+   - Introduces new code without breaking the API
+   - New alternative APIs
+   - Allows the code to compute new things
+   - Quality of life improvements (e.g., extensive optimization)
+   - Entirely new documentation sections
+
+#. Patch
+
+   - Largely modifies existing code
+   - Bug fixes, typos, documentation tweaks
+   - Performance optimizations spanning a couple lines of code
+
+In general the PR considerations are going to be highly tied to what the PR
+wants to accomplish, the following subsections group considerations by use cases
+and assume archetypal PRs. In other words, not all points are going to be
+relevant to every PR. If they're not relevant for a specific PR they should be
+ignored.
+
+General Considerations
+======================
+
+#. Making a PR should be easy.
+
+   - If it's too hard to make a PR people won't do it.
+   - Setting the bar too high discourages new contributors.
+
+#. Should be possible to open the PR early
+
+   - Channel for discussion between reviewers and the PR author.
+   - Gets work on radar of reviewers and other developers.
+   - Time to hash out what is and is not in scope before time is wasted
+
+#. PR lifetimes should be short.
+
+   - The code base moves fast, PRs get stale quickly.
+   - Shoot for merging in less than two weeks.
+   - Merge function by function if necessary.
+
+#. Code submitted as PRs should adhere to the organization's standards.
+
+   - Important for continuity
+   - Makes code reviews easier
+   - Current standards:
+
+      - :ref:`cxx_conventions`.
+      - :ref:`python-coding-conventions`.
+      - :ref:`rest_conventions`.
+      - :ref:`doxygen_conventions`.
+
+#. PRs should contain code of high-caliber
+
+   - NWChemEx strives to be an exemplar package
+   - Easier to maintain good code
+   - High-quality is especially important for senior developers as new
+     developers look to your code for examples.
+   - There is a time and place for "just get something working", but should
+     ideally be avoided
+
+API Breaking Changes
+====================
+
+#. Should be last resort.
+
+   - Maintaining stable APIs leads to users instilling trust in us.
+   - Need to document what was tried to avoid the break.
```

Comment:
I won't make changes throughout the document, but it would be good to be consistent with using periods at the end of bullets or not using bullets.  Maybe a semi-colon?

If we want to go with periods at the end, then we should make everything a correct sentence.  I can make these changes if we go that route.

Reply:
I worked on this, but feel free to change it if you don't like what I have.